File: T4/T4_ViewPager/app/src/main/java/com/system/develop/t4_viewpager/fragments/FragmentUno.java
package com.system.develop.t4_viewpager.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.system.develop.t4_viewpager.R;

public class FragmentUno extends Fragment{

    View v;
    Button b;
    OnFragmentUnoListener listener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        listener = (OnFragmentUnoListener) context;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        v = inflater.inflate(R.layout.fracment_uno_layout,container,false);
        b = v.findViewById(R.id.botonf1);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onFragmentUnoSelected();
            }
        });
        return v;
    }

    public interface OnFragmentUnoListener{
        public void onFragmentUnoSelected();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }
}

File: Kotlin/T5/codigos_ces/T6_Fragments/app/src/main/java/com/example/t6_fragments/ui/fragments/FragmentUno.kt
package com.example.t6_fragments.ui.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.t6_fragments.databinding.FragmentUnoBinding
import com.google.android.material.snackbar.Snackbar

class FragmentUno : Fragment() {

    private lateinit var contexto: Context
    private lateinit var binding: FragmentUnoBinding
    private lateinit var listener: OnFragmentUnoListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.contexto = context
        this.listener = context as OnFragmentUnoListener
        // iniciar las interfaces de callback
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentUnoBinding.inflate(inflater, container, false)
        return binding.root
    }


    override fun onStart() {
        super.onStart()
        binding.botonComunicar.setOnClickListener {
            listener.onNombreSelected(binding.editNombre.text.toString())
            //binding.editNombre.text.clear()
        //Snackbar.make(binding.root, binding.editNombre.text.toString(), Snackbar.LENGTH_SHORT)
            //    .show()
        }

    }

    interface OnFragmentUnoListener{
        fun onNombreSelected(nombre: String)
    }

    override fun onDetach() {
        super.onDetach()
        // eliminar cualquier dependencia
    }

}